### PR TITLE
redhat_subscription: Add support for Red Hat API token

### PR DESCRIPTION
**Pull request by ecchong**
##### SUMMARY
Add support using Red Hat API token

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Updated redhat_subscription.py to allow using Red Hat API token.

##### ADDITIONAL INFORMATION
Added support for using Red Hat API token in addition to activationkey/org ID or username/password combination.
Following this to generate an access token from the offline token.
https://access.redhat.com/articles/3626371

### CODE REVIEW

File: plugins/modules/redhat_subscription.py [hunk 1]
```diff
@@ -40,6 +40,10 @@
         description:
             - access.redhat.com or Red Hat Satellite or Katello password
         type: str
+    token:
+        description:
+            - sso.redhat.com API access token
```
Comment (felixfontein) [inline plugins/modules/redhat_subscription.py:45]:
```suggestion
            - sso.redhat.com API access token.
```

File: plugins/modules/redhat_subscription.py [hunk 2]
```diff
@@ -40,6 +40,10 @@
...
+        type: str
```
Comment (felixfontein) [inline plugins/modules/redhat_subscription.py:46]:
```suggestion
        type: str
        version_added: 6.2.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

**Comment by felixfontein** [comment]
CC @ptoscano

**Comment by ptoscano** [comment]
In principle, this addition sounds OK; I have a couple of notes:
- please squash your commits: even if it will be done at merging time, the history of "do change"/"fix change"/etc is not exactly helpful
- please add at least one test case for this: see the `TEST_CASES` list in `tests/unit/plugins/modules/test_redhat_subscription.py`

Out of curiosity: how were you able to test this? I know that the Keycloak registration does not work OOTB with subscription.rhsm.redhat.com (see https://bugzilla.redhat.com/show_bug.cgi?id=2115893), and it is not enabled in Satellite.

**Comment by felixfontein** [comment]
This PR currently does multiple modifications in unrelated files that must be undone. Also please note the comments by the bot.

**Comment by ecchong** [comment]
> 

It is tested with RHEL8 agains Customer Portal Subscription Management.  I am not sure if Satellite support token yet.
The subscription-manager on RHEL9 still having issue with token, but got a different error.  Will have to look into it and add test cases.

File: changelogs/fragments/5714-proxmox-lxc-tag-support.yml [hunk 3]
```diff
@@ -1,3 +0,0 @@
----
-minor_changes:
-- proxmox - added new module parameter ``tags`` for use with PVE 7+ (https://github.com/ansible-collections/community.general/pull/5714).
```
Comment (felixfontein) [inline changelogs/fragments/5714-proxmox-lxc-tag-support.yml:3]: This file should not be removed. Same for several other changes in this PR.

File: plugins/modules/redhat_subscription.py [hunk 4]
```diff
@@ -40,6 +40,11 @@
         description:
             - access.redhat.com or Red Hat Satellite or Katello password
         type: str
+    token:
+        description:
+            - sso.redhat.com API access token.
+        type: str
+        version_added: 6.2.0
```
Comment (felixfontein) [inline plugins/modules/redhat_subscription.py:47]: Sorry, one last formality:
```suggestion
        version_added: 6.3.0
```

**Comment by felixfontein** [comment]
Looks good from my POV. If @ptoscano is happy I'll merge.

**Review by ptoscano — APPROVED** [review]
Thanks!

**Comment by felixfontein** [comment]
@ecchong thanks for implementing this!
@ptoscano thanks for reviewing!

**Pull request merged**